src/main/java/org/opensearch/knn/common/featureflags/KNNNativeQueryLauncher.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.knn.common.featureflags;
+
+import lombok.experimental.UtilityClass;
+import org.apache.lucene.search.Query;
+import org.opensearch.knn.index.query.KNNQuery;
+import org.opensearch.knn.index.query.nativelib.NativeEngineKnnVectorQuery;
+
+import static org.opensearch.knn.common.featureflags.KNNFeatureFlags.isKnnLaunchQueryRewriteEnabled;
+
+/**
+ * Utility class to decide on whether to use NativeEngineKnnVectorQuery based on a feature flag
+ */
+@UtilityClass
+public class KNNNativeQueryLauncher {
```

Comment:
Do we really need this class , it feels like wrapping isKnnLaunchQueryRewriteEnabled() over another class , 
on the other hand at caller side , it could be easily 
`return isKnnLaunchQueryRewriteEnabled() ? NativeEngineKnnVectorQuery(knnQuery): knnQuery`

I feel this class can be avoided.  It takes burden to write unit test for same thing again over isKnnLaunchQueryRewriteEnabled which was already unit tested

Reply:
> I feel this class can be avoided. It takes burden to write unit test for same thing again ove

The class is for reusability in case we end up adding radial with this. I don't think a small static class is a burden and as far as unit tests are concerned its already written. This class will be gone by 2.19, so I unless there is a valid reason to remove I will prefer to keep it this way

> it could be easily
return isKnnLaunchQueryRewriteEnabled() ? NativeEngineKnnVectorQuery(knnQuery): knnQuery

And we don't have to unit test this?